api/src/main/java/com/datatorrent/api/StatsListener.java

```diff
@@ -115,6 +117,55 @@
     List<OperatorResponse> getOperatorResponse();
   }
 
+  /**
+   * An interface to the DAG. Stats listener can get information about
+   * operator or other elements in the DAG through this interface. currerntly
+   * we only provide method to extract the operator name based on the physical
+   * id of the operator. In future more methods can be added.
+   *
+   */
+  interface StatsListenerContext
```

Comment:
It will be additional interface between DAG and statlistener. It can be used to get additional information about the operator/dag when processStats is called. currently processStats BatchedOperatorStats argument is very limited, in case of shared stats listener it is difficult to know for which operator processStats is being called. In presence of context stats listener can get operator name from operator id to determine the operator. Also it can provide additional interaction between engine and statslistener such as requesting dag change or shutting down the application.